service/src/main/scala/org/apache/celeborn/server/common/http/HttpEndpoint.scala

```diff
@@ -0,0 +1,218 @@
+ */
+
+package org.apache.celeborn.server.common.http
+
+import org.apache.celeborn.server.common.{HttpService, Service}
+
+/**
+ * HTTP endpoints of Rest API providers.
+ */
+trait HttpEndpoint {
+  def path: String
+
+  def meaning(service: String): String
+
+  def handle(service: HttpService, parameters: Map[String, String]): String
+}
+
+case object Conf extends HttpEndpoint {
+  override def path: String = "/conf"
+
+  override def meaning(service: String): String = s"List the conf setting of the $service."
+
+  override def handle(service: HttpService, parameters: Map[String, String]): String =
+    service.getConf
+}
+
+case object WorkerInfo extends HttpEndpoint {
+  override def path: String = "/workerInfo"
+
+  override def meaning(service: String): String = {
+    if (service == Service.MASTER)
+      "List worker information of the service. It will list all registered workers 's information."
+    else "List the worker information of the worker."
+  }
+
+  override def handle(service: HttpService, parameters: Map[String, String]): String =
+    service.getWorkerInfo
+}
+
+case object ThreadDump extends HttpEndpoint {
+  override def path: String = "/threadDump"
+
+  override def meaning(service: String): String = s"List the current thread dump of the $service."
+
+  override def handle(service: HttpService, parameters: Map[String, String]): String =
+    service.getThreadDump
+}
+
+case object Shuffles extends HttpEndpoint {
+  override def path: String = "/shuffles"
+
+  override def meaning(service: String): String = {
+    if (service == Service.MASTER)
+      "List all running shuffle keys of the service. It will return all running shuffle's key of the cluster."
+    else
+      "List all the running shuffle keys of the worker. It only return keys of shuffles running in that worker."
+  }
+
+  override def handle(service: HttpService, parameters: Map[String, String]): String =
+    service.getShuffleList
+}
+
+case object ListTopDiskUsedApps extends HttpEndpoint {
+  override def path: String = "/listTopDiskUsedApps"
+
+  override def meaning(service: String): String = {
+    if (service == Service.MASTER)
+      "List the top disk usage application ids. It will return the top disk usage application ids for the cluster."
+    else
+      "List the top disk usage application ids. It only return application ids running in that worker."
+  }
+
+  override def handle(service: HttpService, parameters: Map[String, String]): String =
+    service.listTopDiskUseApps
+}
+
+case object Help extends HttpEndpoint {
+  override def path: String = "/help"
+
+  override def meaning(service: String): String =
+    s"List the available API providers of the $service."
+
+  override def handle(service: HttpService, parameters: Map[String, String]): String =
+    HttpUtils.help(service.serviceName)
+}
+
+case object Invalid extends HttpEndpoint {
+
+  val invalid = "invalid"
+
+  override def path: String = None.toString
```

Comment:
nit: may `/invalid` instead of `none`?

Reply:
@cfmcgrady, there is no path previously for invalid which is `case _ => "invalid"`, therefore this uses `none` instead of `/invalid`.